Exception when specifying QUARKUS_OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=grpc

### Describe the bug

When using the environment variable `QUARKUS_OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=grpc` Quarkus will throw exception 
```
Caused by: java.lang.IllegalStateException: Only the GRPC Exporter is currently supported. Please check `otel.exporter.otlp.traces.protocol` property
	at io.quarkus.opentelemetry.runtime.exporter.otlp.OtlpRecorder.createOtlpGrpcSpanExporter(OtlpRecorder.java:110)
	at io.quarkus.opentelemetry.runtime.exporter.otlp.OtlpRecorder.installBatchSpanProcessorForOtlp(OtlpRecorder.java:61)
```

I think this issue has been resolved in https://github.com/quarkusio/quarkus/pull/35156 for 3.3.0, but for the older version, this is still causing issues. Can we implement a fix for the older versions? 

Looks like the logic at https://github.com/quarkusio/quarkus/blob/cca68dd37170d3d389fcf7117455ee798f6d055e/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/OtlpRecorder.java#L109 is a little bit wrong. Shouldn't the error be thrown when the protocol not equals to `grpc` instead? 

Thanks! Please feel free to correct me if I am getting any thing wrong here.  

### Expected behavior

No exception should be thrown

### Actual behavior

Exception thrown 

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/exporter/otlp/OtlpExporterConfigTest.java
+++ b/extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/exporter/otlp/OtlpExporterConfigTest.java
@@ -16,7 +16,7 @@ public class OtlpExporterConfigTest {
     static final QuarkusUnitTest TEST = new QuarkusUnitTest()
             .withEmptyApplication()
             .overrideConfigKey("otel.traces.exporter", "cdi")
-            .overrideConfigKey("otel.exporter.otlp.traces.protocol", "http/protobuf")
+            .overrideConfigKey("otel.exporter.otlp.traces.protocol", "grpc")
             .overrideConfigKey("quarkus.opentelemetry.tracer.exporter.otlp.endpoint", "http://localhost ")
             .overrideConfigKey("quarkus.otel.bsp.schedule.delay", "50")
             .overrideConfigKey("quarkus.otel.bsp.export.timeout", "1s");

--- a/extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/exporter/otlp/OtlpExporterUnsupportedProtocolTest.java
+++ b/extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/exporter/otlp/OtlpExporterUnsupportedProtocolTest.java
@@ -1,0 +1,47 @@
+package io.quarkus.opentelemetry.deployment.exporter.otlp;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class OtlpExporterUnsupportedProtocolTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .withEmptyApplication()
+            .overrideConfigKey("otel.traces.exporter", "cdi")
+            .overrideConfigKey("quarkus.otel.exporter.otlp.traces.protocol", "http/protobuf")
+            .overrideConfigKey("quarkus.opentelemetry.tracer.exporter.otlp.endpoint", "http://localhost ")
+            .overrideConfigKey("quarkus.otel.bsp.schedule.delay", "50")
+            .overrideConfigKey("quarkus.otel.bsp.export.timeout", "1s")
+            .assertException(t -> {
+                Throwable e = t;
+                IllegalStateException ie = null;
+                while (e != null) {
+                    if (t instanceof IllegalStateException) {
+                        ie = (IllegalStateException) t;
+                        break;
+                    }
+                    e = e.getCause();
+                }
+
+                if (ie == null) {
+                    fail("No IllegalStateException thrown: " + t);
+                }
+                assertTrue(ie.getMessage().contains("Only the `grpc` protocol is currently supported."),
+                        ie.getMessage());
+                assertTrue(ie.getMessage().contains("`http/protobuf` is available after Quarkus 3.3."),
+                        ie.getMessage());
+                assertTrue(ie.getMessage().contains("Please check `otel.exporter.otlp.traces.protocol` property"),
+                        ie.getMessage());
+            });
+
+    @Test
+    void testNotSupportedProtocolConfig() {
+        fail();
+    }
+}

--- a/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/config/runtime/exporter/OtlpExporterTracesConfig.java
+++ b/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/config/runtime/exporter/OtlpExporterTracesConfig.java
@@ -69,8 +69,10 @@ public interface OtlpExporterTracesConfig {
     /**
      * OTLP defines the encoding of telemetry data and the protocol used to exchange data between the client and the
      * server. Depending on the exporter, the available protocols will be different.
+     * <p>
+     * Currently, only {@code grpc} is allowed.
      */
-    @WithDefault(Protocol.HTTP_PROTOBUF)
+    @WithDefault(Protocol.GRPC)
     Optional<String> protocol();
 
     public static class Protocol {

--- a/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/OtlpRecorder.java
+++ b/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/OtlpRecorder.java
@@ -2,7 +2,7 @@ package io.quarkus.opentelemetry.runtime.exporter.otlp;
 
 import static io.quarkus.opentelemetry.runtime.OpenTelemetryUtil.convertKeyValueListToMap;
 import static io.quarkus.opentelemetry.runtime.config.runtime.exporter.OtlpExporterRuntimeConfig.DEFAULT_GRPC_BASE_URI;
-import static io.quarkus.opentelemetry.runtime.config.runtime.exporter.OtlpExporterTracesConfig.Protocol.HTTP_PROTOBUF;
+import static io.quarkus.opentelemetry.runtime.config.runtime.exporter.OtlpExporterTracesConfig.Protocol.GRPC;
 
 import java.util.List;
 import java.util.Map;
@@ -106,8 +106,9 @@ public class OtlpRecorder {
                     }
                 });
 
-        if (!exporterRuntimeConfig.traces().protocol().orElse("").equals(HTTP_PROTOBUF)) {
-            throw new IllegalStateException("Only the GRPC Exporter is currently supported. " +
+        if (!exporterRuntimeConfig.traces().protocol().orElse("").equals(GRPC)) {
+            throw new IllegalStateException("Only the `grpc` protocol is currently supported. " +
+                    "`http/protobuf` is available after Quarkus 3.3. " +
                     "Please check `otel.exporter.otlp.traces.protocol` property");
         }
         return exporterBuilder.build();

